expression/builtin_regexp_util.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"regexp"
+	"sync"
+
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+// Parameters may be const or ignored by the user, so different situations should be considered
+// We can handle parameters more easily with this struct.
+//
+// notProvided field shows if user ignores this param in sql
+// for example:
+//
+//	select regexp_like("123", "123", "m"), here notProvided field is false for the third parameter
+//	select regexp_like("123", "123"), here notProvided field is true for the third parameter
+type regexpParam struct {
+	constStrVal   string
+	constIntVal   int64
+	defaultIntVal int64 // default value when notProvided is true
+	isConst       bool
+	notProvided   bool
+	col           *chunk.Column
```

Comment:
Can we remove `notProvided` and use `re.col == nil` instead?

Reply:
> Can we remove `notProvided` and use `re.col == nil` instead?

I think providing a new field is easier to understand the code.